app/controllers/middleware_deployment_controller.rb

```diff
@@ -17,4 +28,48 @@ def show
     show_container(@record, controller_name, display_name)
   end
 
+  def button
+    selected_operation = params[:pressed].to_sym
+    if OPERATIONS.key?(selected_operation)
```

Comment:
Reading below I think that you can create a class under `app/services/`. You have logic that deals with the various operations that you can put there. But you use `params` and `flash` in that so that would be better left in the controller.

Anyway you can work on that in a separate PR.